apps/aechannel/src/aesc_fsm.erl

```diff
@@ -2744,11 +2882,17 @@ maybe_check_sigs(Tx, Updates, TxType, WrongTxTypeMsg, Who, NextState, D)
             keep_state(D)
     end.
 
-pubkeys(Who, D) ->
-    case Who of
-        me -> [my_account(D)];
-        other_participant -> [other_account(D)];
-        both -> both_accounts(D)
+pubkeys(both, D, _) ->
+    both_accounts(D);
+pubkeys(Who, D, SignedTx) ->
+    case aesc_utils:count_authentications(SignedTx) of
```

Comment:
Looking at the implementation of ```aesc_utils:count_authentications/1``` function this will fail if a TX is wrapped in multiple MetaTx from Alice and Bob - for instance AliceMeta1(AliceMeta2(BobMeta1(BobMeta2(Tx))))